internal/build/archive.go

```diff
@@ -272,13 +272,18 @@ func extractFile(arpath string, armode os.FileMode, data io.Reader, dest string)
 		return fmt.Errorf("path %q escapes archive destination", target)
 	}
 
-	// Ensure the destination directory exists.
+	// Remove the preivously-extracted file if it exists
+	if err := os.RemoveAll(filepath.Dir(target)); err != nil {
```

Comment:
I had to make this change because calling `open` with `O_TRUNC` will fail if the file does not have user or group permissions (the `readme.txt` in the `protoc` directory triggers this).

[source](https://pubs.opengroup.org/onlinepubs/9699919799/functions/open.html): `open` will return `EACCES` if 
> O_TRUNC is specified and write permission is denied.